core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt

```diff
@@ -51,9 +53,36 @@ fun View.snack(
   ).apply {
     actionStringId?.let { setAction(it) { actionClick?.invoke() } }
     setActionTextColor(actionTextColor)
+    anchorView = anchor
+    addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
+      override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
+        transientBottomBar?.anchorView = null
+      }
+    })
   }.show()
 }
 
+@SuppressLint("ShowToast")
+fun View.snack(
+  stringId: Int,
+  anchor: View,
+  actionStringId: Int? = null,
+  actionClick: (() -> Unit)? = null,
+  @ColorInt actionTextColor: Int = Color.WHITE
+) {
+  Snackbar.make(
+    this, stringId, Snackbar.LENGTH_LONG
+  ).apply {
+    actionStringId?.let { setAction(it) { actionClick?.invoke() } }
+    setActionTextColor(actionTextColor)
+    anchorView = anchor
+    addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
+      override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
+        transientBottomBar?.anchorView = null
+      }
+    })
+  }.show()
+}
```

Comment:
Please follow the proper code styling, and fix the whitespace error here.
![Screenshot from 2023-04-28 12-43-28](https://user-images.githubusercontent.com/34593983/235081648-673f4490-26df-4865-bdc8-ea83098d6797.png)